controllers/front/ProductController.php

```diff
@@ -954,6 +955,25 @@ protected function formatQuantityDiscounts($specific_prices, $price, $tax_rate,
         return $specific_prices;
     }
 
+    /**
+     * @param float $specificPriceValue
+     * @param mixed $specificPriceReductionTax
+     * @param float $ecoTaxAmount
+     *
+     * @return mixed
+     */
+    private function calculatePriceDefaultCurrency($specificPriceValue, $specificPriceReductionTax, $taxRate, $ecoTaxAmount)
+    {
+        $priceDefaultCurrency = $specificPriceValue;
+
+        // only apply tax rate when tax calculation method is set to PS_TAX_INC
+        if (Product::$_taxCalculationMethod == PS_TAX_INC && $specificPriceReductionTax) {
```

Comment:
```suggestion
        if (Product::$_taxCalculationMethod === PS_TAX_INC && $specificPriceReductionTax) {
```

Reply:
This is legacy code that I extracted in a function...  I really don't know if Product::$_taxCalculationMethod will always be an integer (and not, let's say, a string).

It felt safer to keep the legacy behavior in this condition. I'll check that out.

Reply:
This value can only be an integer (PS_TAX_EXC and PS_TAX_INC) / null (intiliazed as property):
Feel free to decide :sweat_smile:

Reply:
Ok, from what I checked in the codebase, it seems that this variable is always assigned an integer. Done.